File: src/main/java/me/aikoo/Karl/Interpreter/parser/ast/values/CharValue.java
package me.aikoo.Karl.Interpreter.parser.ast.values;

import me.aikoo.Karl.Interpreter.parser.TokenType;

public class CharValue extends Value {
    private final char value;

    public CharValue(char value) {
        this.value = value;
    }

    public char getValue() {
        return value;
    }

    @Override
    public String toString() {
        return String.valueOf(value);
    }

    @Override
    public int toInt() {
        return Integer.parseInt(toString());
    }

    @Override
    public float toFloat() {
        return Float.parseFloat(toString());
    }

    @Override
    public TokenType getType() {
        return TokenType.CHAR_VALUE;
    }
}

File: src/main/kotlin/studio/karllang/karl/parser/ast/values/CharValue.kt
package studio.karllang.karl.parser.ast.values

import studio.karllang.karl.lexer.TokenType

class CharValue(val value: Char) : Value() {
    override val type: TokenType = TokenType.CHAR_VALUE

    override fun toString(): String {
        return value.toString()
    }

    override fun toInt(): Int {
        return toString().toInt()
    }

    override fun toFloat(): Float {
        return toString().toFloat()
    }
}